### ISSUE TITLE
Support for alternative search engines

### ISSUE THREAD

**Issue by matchatype** [FusionAuth/fusionauth-issues#1256]
I'm struggling to define decent initial requirements to start up FusionAuth in production, with most of my problems regarding the deployment of an ES cluster. If it wasn't for ES, FusionAuth would be a lot easier to start with. With thousands config options, ES looks way more complex and resource intensive than FusionAuth itself.

### Solution
Add support for an alternative engine that's easier and lighter than ElasticSearch, like [Typesense](https://typesense.org) or [Meilisearch](https://www.meilisearch.com). This would shorten the gap between DB search and ES.

### Alternatives/workarounds
None. Currently available DB search capabilities are too limited.

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
We have so much invested in Elasticsearch, there would have to be a huge value to FusionAuth to supporting a third option for search. 

Lots of options:

1. Use docker, super simple.  `docker run -p 9021:9200 -e 'discovery.type=single-node' -e 'http.publish_host=localhost' -e 'http.publish_port=9021' docker.elastic.co/elasticsearch/elasticsearch:7.12.1`
2. Use the FusionAuth search bundle which hides all of the complexity. Super simple. See `fusionauth-search` package in the install guide. 
3. If  you don't want to hassle with it at  all in prod, you can use any of the many managed ES services available so all of the complexity is hidden. 
4. Or us the db search option if you don't want to deal with any of this, that is what that is for, it of course comes with a huge downside of less than powerful search options.

**Comment by matchatype** [comment]
I agree with @robotdan in that ES integration with FusionAuth is seamless, and installation is definitely unproblematic. The business case for a third option for search is only reduced operational burden and infrastructure costs, considering there are other options to achieve optimized sub-50ms searches that do not require the extra resources that ES needs, while providing an intuitive developer experience.

**Comment by mooreds** [comment]
I'm closing because I don't think we're going to do this in the foreseeable future.
